contracts/core/Dispatcher.sol

```diff
@@ -24,25 +36,19 @@ contract Dispatcher is IbcDispatcher, IbcEventsEmitter, Ownable, Ibc {
     uint32 portPrefixLen;
 
     mapping(address => mapping(bytes32 => Channel)) public portChannelMap;
-    mapping(address => mapping(bytes32 => uint64)) nextSequenceSend;
     // keep track of received packets' sequences to ensure channel ordering is enforced for ordered channels
-    mapping(address => mapping(bytes32 => uint64)) nextSequenceRecv;
-    mapping(address => mapping(bytes32 => uint64)) nextSequenceAck;
+    mapping(address => mapping(bytes32 => PackedNextSequence)) nextSequence;
     // only stores a bit to mark packet has not been ack'ed or timed out yet; actual IBC packet verification is done on
     // Polymer chain.
-    // Keep track of sent packets
-    mapping(address => mapping(bytes32 => mapping(uint64 => bool))) public sendPacketCommitment;
-    // keep track of received packets to prevent replay attack
-    mapping(address => mapping(bytes32 => mapping(uint64 => bool))) recvPacketReceipt;
-    // keep track of outbound ack packets to prevent replay attack
-    mapping(address => mapping(bytes32 => mapping(uint64 => bool))) ackPacketCommitment;
+    // keep track of packets to prevent replay attack
+    mapping(address => mapping(bytes32 => mapping(uint64 => PackedCommits))) public packetCommitment;
 
     ConsensusStateManager consensusStateManager;
 
     //
     // methods
     //
-    constructor(string memory initPortPrefix, ConsensusStateManager _consensusStateManager) {
+    constructor(string memory initPortPrefix, ConsensusStateManager _consensusStateManager) payable {
```

Comment:
```suggestion
    constructor(string memory initPortPrefix, ConsensusStateManager _consensusStateManager) {
```

Actually, I think we should leave this payable out because the solidity check that msg.value = 0 is useful for ensuring that we don't accidentally send any tokens to this during deployment. I'm mainly thinking of the case where something is wrong in our deployment script and we accidentally pass in a value for this consturctor deployment - all the tokens would be lost in this contract if that happens. Granted this has a very small likelyhood of happening, but imo it's worth the small gas cost.

Also this gas optimization would only impact deployment cost from our end, and not dapp devs who are using this so I think it's fine for us to pay this additional gas cost to insure we don't send any tokens to this contract that would be lost.